### PR TITLE
Function with abstract axis (no recompilation) can be executed multiple times

### PR DESCRIPTION
**Pull request by rmoyard**
**Context:**
A qjitted function with abstracted axis that is already compiles fails the recompilation checks when executed again.

**Description of the Change:**
- Remove empty dictionnary in a Jax function
- Add test for execution and no recompilation

Closes https://github.com/PennyLaneAI/catalyst/issues/1231
[sc-76820]

### CODE REVIEW

**Comment by rmoyard** [comment]
Closes https://github.com/PennyLaneAI/catalyst/issues/1231

File: frontend/test/pytest/test_jax_dynamic_api.py [hunk 1]
```diff
@@ -1139,5 +1139,48 @@ def f(i, _):
...
+    @qml.qjit(abstracted_axes=(("n",), ()))
+    @qml.qnode(qml.device("lightning.qubit", wires=2))
+    def circuit(x1, x2):
+
+        @qml.for_loop(0, jnp.shape(x1)[0], 1)
+        def loop_block(i):
+            qml.Hadamard(0)
+            qml.RX(x1[i], 0)
+            qml.CNOT(wires=[0, 1])
+            qml.Hadamard(1)
```
Comment (paul0403) [inline frontend/test/pytest/test_jax_dynamic_api.py:1154]: To save testing time in CI, why don't we remove the static gates that don't depend on the arguments?

Comment (rmoyard) [reply frontend/test/pytest/test_jax_dynamic_api.py]: Done :+1:

**Review by paul0403 — APPROVED** [review]
Thanks!
